tests/test_pytest.py

```diff
@@ -0,0 +1,29 @@
+import subprocess
+
+import dbusmock
+
+
+def test_dbusmock_testcase_spawn_server(dbusmock_testcase_session):
+    test_iface = 'org.freedesktop.Test.Main'
+
+    p_mock = dbusmock_testcase_session.spawn_server(
+        'org.freedesktop.Test', '/', test_iface, stdout=subprocess.DEVNULL)
+
+    obj_test = dbusmock_testcase_session.get_dbus().get_object('org.freedesktop.Test', '/')
+
+    obj_test.AddMethod('', 'Upper', 's', 's', 'ret = args[0].upper()', interface_name=dbusmock.MOCK_IFACE)
+    assert obj_test.Upper('hello', interface=test_iface) == 'HELLO'
+
+    p_mock.terminate()
+    p_mock.wait()
```

Comment:
This sucks. While it passes without the cleanup, that'd be relying too much on process exit behaviour, and may leak too many resources. Ideally spawn_server() can clean this up by itself, but it's class-scoped, so no `addCleanup()`.

I deliberately want to keep the fixtures session-scoped, as opposed to starting a new D-Bus for each single test (which is slow and noisy).

Reply:
I think what you want in most cases here is abstract the mock through a local fixture, see #182 where I did this - the tests then use that mock as argument and run the actual test, you need less in-test teardown.

At least that's the best use I've found for the xdg-desktop-portal tests.

Reply:
Thanks! This is nicer indeed, done in one test case to illustrate it.